torch_geometric/transforms/add_node_feature.py

```diff
@@ -0,0 +1,56 @@
+from typing import Tuple, Union
+
+import torch
+from torch import Tensor
+
+from torch_geometric.data import Data, HeteroData
+from torch_geometric.data.datapipes import functional_transform
+from torch_geometric.transforms import BaseTransform
+
+
+@functional_transform('add_node_feature')
+class AddNodeFeature(BaseTransform):
```

Comment:
This feels quite related to `transforms.Constant`. Can we unify them?

Reply:
Sounds great. But `Constant` does not support adding random features, as introduced as a trick in [Random Features Strengthen Graph Neural Networks](https://arxiv.org/abs/2002.03155).

Reply:
But that probably deserves its own transform anyway (with a proper cite to the paper). WDYT?

Reply:
Agreed. I'm gonna unify them into `transforms.Constant`.

Reply:
@rusty1s Will that make sense now?